tests/test_storage.py

```diff
@@ -55,8 +57,41 @@ def test_merged_storage(self):
             assert_allclose(b.timestamps, a.timestamps)
             assert_allclose(b.feature, a.feature)
 
+    def test_out_of_order(self):
+        features = [
+            self.fake_vf("Q000002", 10),
+            self.fake_vf("Q000003", 20, fps=3.0),
+            self.fake_vf("Q000001", 30, fps=0.5),
+        ]
+        with tempfile.NamedTemporaryFile() as f:
+            store_features(f, features, Dataset.QUERIES)
+            f.flush()
+            data = np.load(f.name, allow_pickle=False)
+
+        # Scramble the order.
+        N = data["features"].shape[0]
+        order = np.random.permutation(N)
+        data = {
+            "features": data["features"][order, :],
+            "video_ids": data["video_ids"][order],
+            "timestamps": data["timestamps"][order],
+        }
+        with tempfile.NamedTemporaryFile() as f:
+            np.savez(f, **data)
+            f.flush()
+            restored = load_features(f.name)
+
+        features.sort(key=lambda x: x.video_id)
+        restored.sort(key=lambda x: x.video_id)
```

Comment:
Should we be testing that `restored` is already properly sorted when loading with `load_features`? I'm not sure we should sort it here.